src/formatDate.js

```diff
@@ -7,8 +7,37 @@
  *
  * @returns {string}
  */
-function formatDate(date, fromFormat, toFormat) {
-  // write code here
+function formatDate(dateString, fromFormat, toFormat) {
+  const separator = fromFormat[3];
+  const newSeparator = toFormat[3];
+  const dateParts = dateString.split(separator);
+
+  const dateObj = {
+    YYYY: fromFormat.includes('YYYY')
+      ? dateParts[fromFormat.indexOf('YYYY')]
+      : null,
+    YY: fromFormat.includes('YY') ? dateParts[fromFormat.indexOf('YY')] : null,
+    MM: dateParts[fromFormat.indexOf('MM')],
+    DD: dateParts[fromFormat.indexOf('DD')],
+  };
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    const year = +dateObj['YY'];
+
+    dateObj['YYYY'] = year < 30 ? `20${dateObj['YY']}` : `19${dateObj['YY']}`;
+  }
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    dateObj['YY'] = dateObj['YYYY'].slice(-2);
```

Comment:
Similarly, the logic for converting from YYYY to YY is correct. However, as with the previous comment, consider extracting this into a separate function for better modularity and readability.